### ISSUE TITLE
[Request] Remove analytics or give option to disable analytics

### ISSUE THREAD

**Issue by cukoid** [AVnetWS/Hentoid#92]
^

**Comment by avluis** [comment]
Any particular reason why?
We sort of need the data since the community is pretty quiet about things (minus a few here and there) - I still don't have my testers.
Also, what happened to that template there, what is this: ^?

**Comment by wightwulf1944** [comment]
I also would like to know the specific concern here. Can you tell us about the motivation behind this?

**Comment by cukoid** [comment]
Because el' Google doesn't need to know when I'm fappin' to Hentai.

Also, if you guys ever tried to get into the F-Droid repos, they would not approve it mostly because of Google Analytics or they would mark it as an antifeature.

**Comment by avluis** [comment]
LOL!
Man, this is too funny - Google does not know when you fappin'. LOL.

The reason I mentioned the " ^ " is because this repo has templates that need to be filled out before an issue is submitted, which you simply ignored.

Give me a proper reason why Google Analytics should not be on this application, and not a big brother is watching sort of reason.

**Comment by cukoid** [comment]
Because you should use a Free and Open Sauce analytics software like Piwik instead. Or give a toggle in settings to disable it.

**Comment by avluis** [comment]
Hold on - so you don't have no problem with a bit of data being collected, as long as it is not Google?

**Comment by cukoid** [comment]
Yes, or if you give the option to disable tracking.

**Comment by avluis** [comment]
If I follow this request, I would implement Piwik, leaving Google Analytics alone while giving the user an option to choose between tracking.

**Comment by cukoid** [comment]
I guess that's fine, but I don't understand why you can't have a "no tracking" option, instead of choosing between GA and Piwik.

**Comment by wightwulf1944** [comment]
Hi,

Regarding personally identifiable data, google analytics does not send any of the sort. It is literally impossible to identify who is doing what at any time with the data that is collected by analytics. We opted to choose google analytics because it is a service that is under careful scrutiny of many people. Because of that we can be sure that nothing personal is being collected and no rights are being infringed

If we were to use a different analytics service however we would have to take action to ensure that all data being collected is safe, and does not infringe privacy, and local laws.

tl;dr: We're sure that google analytics is safe, we're not so sure about other services. For more information on this, you may read google analytics' [terms of service found here](http://www.google.com.ph/analytics/terms/us.html)

Regarding your concern about f-droid's inclusion policy. Unfortunately Hentoid violates almost every item in the list. As funny as this may seem, I don't mean this to be a joke. Particularly "trademarks must not be infringed, and any other legal requirements must be adhered to". Though Hentoid does not host any copyrighted works it can be considered as a tool that enables users to download such material.

As such, we no longer hope to have Hentoid available in f-droid's repo.

[F-droid inclusion policy here](https://f-droid.org/wiki/page/Inclusion_Policy)

And finally, since this is an open source project you're more than welcome to walk through the code and see what kind of information exactly is being collected.

**Comment by cukoid** [comment]
I apologize if this wasted your time.

Thanks for the great application.

I am not a programmer, so I'm not sure if/how I can help with "testing", but I can if that doesn't require me to be one, if you want.

**Comment by wightwulf1944** [comment]
Oh no, we find these kinds of discussion to be insightful. And I just want to emphasize that we are not educated in law, so we'd rather take the safe route with google analytics and not be sued for information we didn't know we were collecting.

we'll try to contact you when development resumes and we have things we need testing with. Please don't hesitate to contact us through our fakkudroid blog, google+ community, or file an issue here if you have any further concerns. And please make github issues detailed so that we can resolve your concern in a timely manner.

**Comment by cukoid** [comment]
Um, there are manga readers on F-Droid, so I don't understand how your app would violate F-Droid's terms if those apps don't.

**Comment by avluis** [comment]
@cukoid It's OK - every concern is a valid one.
At the moment, any tester I need does not need to be knowledgeable with programming at all!
I send you a build to test, and if you don't already know, I walk you through the steps required in order to get it installed.

Anyways, addressing a few more points, particularly regarding Piwik:
The self hosting is good (I would need to run a website for it to work), but bad at the same time, if that website is down, then so is tracking.
The self hosting is also good because we can control the data, but also bad because I then need to put time into making sure that this data is not compromised, misused and otherwise taken care of properly.

With Google Analytics, I scrounged every possible setting to make sure no user identifiable data is ever tracked, this is also against their terms of use as @wightwulf1944 mentioned above.
The main reason for Google Analytics is one simple reason:
We know were the app needs work, where it crashes, on what Android version it crashes on and what crashes.

I'm sorry I came off a bit strong, but this data is invaluable to a small project like this, I hope you can understand this.

**Comment by cukoid** [comment]
https://f-droid.org/repository/browse/?fdid=eu.kanade.tachiyomi
https://f-droid.org/repository/browse/?fdid=ar.rulosoft.mimanganu

**Comment by avluis** [comment]
@cukoid I've been aware of those applications being on that repository - believe it or not, it's still a grey area - even worse when dealing with copyrighted doujinshi works, which Hentoid facilitates access to.

**Comment by wightwulf1944** [comment]
I might contact the developers of those apps to see what they've done regarding the policy.

Also, in-app updaters are also not allowed.

**Comment by avluis** [comment]
@wightwulf1944 I forgot to mention that part as well - if we want the updater to stay and F-Droid repo, it won't happen - even if we can somehow stay on there as soon as they realize what Hentoid gives access to.

**Comment by cukoid** [comment]
avluis, I have it, just not installed atm because I'm "behind with the times" on what manga is good. Lol. I'm going to install it again. (Off topic I know)

**Comment by avluis** [comment]
Yeah, I know it's a bit off topic - who cares, LOL!
@wightwulf1944 Please don't kill me.

@cukoid When you do, opt in to the beta version of their app, it is much better.
Once you do, visit the 'All Manga' tab, it should sort by popularity by default.

**Comment by cukoid** [comment]
@avluis  where do i apply to become a tester, the github issue for it is locked

**Comment by avluis** [comment]
@cukoid You would message me directly - but since I know you are interested, I can add you right now.
Is that the case?

**Comment by cukoid** [comment]
Yes, that is the case.

I did submit Hentoid to the F-Droid submission queue subforum, but the problem with it at this point is that it "uses play services". Not sure if you cared about getting it on F-Droid or not (I personally think it would increase users, maybe) but I thought I would mention that regardless.

**Comment by avluis** [comment]
Correct - anything that uses private sources cannot be added to the F-Droid repo.

**Comment by cukoid** [comment]
Moto G 2014 (CM13 - Android 6.0.1)
Nexus 7 2013 (Android 6.0.1)
Shield Portable (Android 4.4.2)

**Comment by avluis** [comment]
@cukoid What was the last official update for that Moto G that you ended up installing CM?
How does it run there, and do you have GApps installed?

**Comment by cukoid** [comment]
@avluis I think it was 5.0.1. But now there's an official 6.0.1.
CM13 runs fine most of the time. When the phone's CPU gets busy, it bogs it down a lot, but that's the same with the official firmware.
And yes, I have Gapps installed.

**Comment by mffe** [comment]
@avluis Can you please add an additional branch that follows the F-Droid inclusion policy? (i.e. no play services and auto-updater)

**Comment by avluis** [comment]
@mffe Who says we want to be on F-Droid?

**Comment by mffe** [comment]
@avluis I know it's your decision to add a F-Droid branch to this repository (which would merely involve removing some lines), but you're missing out on a lot of technically inclined users who may improve this open source app.

**Comment by avluis** [comment]
@mffe What build [release] of Hentoid do you have installed?

**Comment by mffe** [comment]
@avluis I don't have Hentoid installed because of this issue.

**Comment by Nonononoki** [comment]
@avluis Sorry for reviving old issue, but do you mind if I fork this project and try to get it on F-Droid?

**Comment by wightwulf1944** [comment]
@Nonononoki no problem man. But remember that F-Droid policy states that apps that allow access to illegal content is not allowed. I would recommend using a new email for your f-droid account in case it gets banned.

**Comment by Thewisem** [comment]
Its forked and is on fdroid. Nice
